tt_metal/include/tt_metal/device.hpp

```diff
@@ -0,0 +1,284 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cstddef>
+
+#include "tt_metal/impl/buffers/buffer_constants.hpp"
+#include "tt_metal/impl/dispatch/work_executor.hpp"
+#include "types.hpp"
+
+//==================================================
+//               DEVICE MANAGEMENT
+//==================================================
+
+namespace tt::tt_metal {
+namespace v1 {
+
+/**
+ * @brief Returns the number of Tenstorrent devices that can be targeted.
+ *
+ * @return Size_t representing the number of available devices.
+ */
+std::size_t GetNumAvailableDevices();
+
+/**
+ * @brief Returns the number of Tenstorrent devices connected via PCIe.
+ *
+ * @return Size_t representing the number of PCIe devices.
+ */
+std::size_t GetNumPCIeDevices();
+
+/**
+ * @brief Retrieves the PCIe device ID for a given device ID.
+ *
+ * @param device_id ID of the device to query.
+ * @return Chip ID of the PCIe device.
+ */
+chip_id_t GetPCIeDeviceID(chip_id_t device_id);
+
+/**
+ * @brief Instantiates a Device object.
+ *
+ * @param device_id ID of the device to target (0 to GetNumAvailableDevices() - 1).
+ * @param config Device configuration parameters
+ * @return Device handle to the created device.
+ */
+DeviceHandle CreateDevice(chip_id_t device_id, DeviceConfig config);
+
+/**
+ * @brief Resets and closes the device.
+ *
+ * @param device Handle to the device to close.
+ * @return True if the device was successfully closed; otherwise, false.
+ */
+bool CloseDevice(DeviceHandle device);
+
+/**
+ * @brief Deallocates all buffers on the device.
+ */
+void DeallocateBuffers(DeviceHandle device);
+
+/**
+ * @brief Dumps device-side profiler data to a CSV log.
+ *
+ * @param device The device holding the program being profiled.
+ * @param worker_cores CoreRangeSet of worker cores being profiled.
+ * @param last_dump If true, indicates the last dump before process termination.
+ */
+void DumpDeviceProfileResults(DeviceHandle device, const CoreRangeSet &worker_cores, bool last_dump = false);
```

Comment:
maybe outside of the scope of this PR but should this go into a profiler header?